escrow/lib.rs

```diff
@@ -0,0 +1,445 @@
+            assert_eq!(
+                deposit(&mut escrow, buyer, 199, EXPECTED_DEPOSIT),
+                Err(Error::ContractAlreadyCompleted)
+            );
+            assert_eq!(
+                confirm_delivery(&mut escrow, buyer),
+                Err(Error::ContractAlreadyCompleted)
+            );
+            assert_eq!(
+                refund_deposit(&mut escrow, seller),
+                Err(Error::FundsCannotBeReturned)
+            );
+        }
+
+        #[ink::test]
+        fn given_new_escrow_contract_when_deposit_is_not_made_and_delivery_done_then_error_is_returned(
+        ) {
+            let (mut escrow, _, buyer, _) = create_contract(0);
+            set_caller(buyer);
+            assert_eq!(escrow.confirm_delivery(), Err(Error::FundsNotDepositedYet));
+        }
+
+        #[ink::test]
+        fn given_new_escrow_contract_when_delivery_done_as_not_buyer_then_error_is_returned() {
+            let (mut escrow, _, _, seller) = create_contract(0);
+            set_caller(seller);
+            assert_eq!(
+                escrow.confirm_delivery(),
+                Err(Error::ConfirmDeliveryNotAsBuyer)
+            );
+        }
+
+        #[ink::test]
+        fn given_deposited_escrow_contract_when_refund_deposit_is_called_then_deposit_is_returned()
+        {
+            const EXPECTED_DEPOSIT: Balance = 1123;
+            const INITIAL_BUYER_BALANCE: Balance = 2 * EXPECTED_DEPOSIT;
+            // below w/a fact that payable does not work in off-chain settings
+            // so we set contract balance in advance in create_contract
+            // while it should happen in escrow.deposit()
+            let (mut escrow, _, buyer, seller) = create_contract(EXPECTED_DEPOSIT);
+            assert_eq!(
+                deposit(&mut escrow, buyer, INITIAL_BUYER_BALANCE, EXPECTED_DEPOSIT),
+                Ok(())
+            );
+            set_balance(buyer, 0);
+            assert_eq!(refund_deposit(&mut escrow, seller), Ok(()));
+            assert_eq!(escrow.state, State::Completed);
+            assert_eq!(escrow.deposit, 0);
+            assert_eq!(get_balance(buyer), EXPECTED_DEPOSIT);
+            assert_eq!(get_balance(contract_id()), 0);
+        }
+
+        #[ink::test]
+        fn given_not_deposited_escrow_contract_when_refund_deposit_is_called_then_deposit_is_not_returned(
+        ) {
+            // below w/a fact that payable does not work in off-chain settings
+            // so we set contract balance in advance in create_contract
+            // while it should happen in escrow.deposit()
+            let (mut escrow, _, _, seller) = create_contract(10);
+            assert_eq!(
+                refund_deposit(&mut escrow, seller),
+                Err(Error::FundsCannotBeReturned)
+            );
+        }
+
+        #[ink::test]
+        fn given_completed_escrow_contract_when_refund_deposit_is_called_by_not_seller_then_deposit_is_not_returned(
```

Comment:
I think going beyond certain length with names (even test names) is kinda counterproductive. This goes waaaay beyond that limit. These names are neither decriptive nor easy to read or work with

Reply:
done